feat: add global bibliography support to pandoc-reader plugin

filipaldi opened the pull request:
- Add support for global bibliography files with configurable names
- Maintain backward compatibility with individual bibliography files
- Add PANDOC_GLOBAL_BIB_FILES setting
- Add comprehensive tests for all scenarios
- Update documentation and provide example configuration
- Add .venv and __pycache__ to .gitignore

nandac commented:
@filipaldi Thank you for your pull request. I will need to set aside some time to review it.

From what I understand, the feature allows you to add a common bibliography to all your blogs or posts, which is a useful feature to have. However, how would you switch the feature off in cases where you do not want to add the common bibliography?

justinmayer reviewed (changes requested):
Thank you for the submission, @filipaldi. In addition to what @nandac said, I made a comment requesting that the `.gitignore` additions be removed. Also, CI is reporting test/linter failures, but that appears to be due to outdated GitHub Actions that I have since updated via [ddfb7e8](https://github.com/pelican-plugins/pandoc-reader/commit/ddfb7e8b7cd1e9df62316003c04de4fcafccdeda). If you rebase your branch on current `main`, those CI-related issues should hopefully be resolved.

nandac commented:
@filipaldi kindly rebase your branch and make the changes requested by @justinmayer. We will then discuss any changes or improvements before we merge this feature in.

Commit:
feat: add global bibliography support to pandoc-reader plugin

- Add support for global bibliography files with configurable names
- Maintain backward compatibility with individual bibliography files
- Add PANDOC_GLOBAL_BIB_FILES setting
- Add comprehensive tests for all scenarios
- Update documentation and provide example configuration

justinmayer commented:
@filipaldi: It seems the code style linters did not pass in CI. Have you followed the [Plugin Development Docs](https://docs.getpelican.com/en/latest/contribute.html#plugin-development) to ensure that the linters pass locally?

Commit:
Fix linting issues: remove trailing whitespace and unused imports

nandac commented on README.md:
```diff
@@ -181,9 +181,31 @@ Without these settings, citations will not be processed by the plugin.
+* `bibliography.bib`, `bibliography.json`, `bibliography.yaml`, `bibliography.bibtex`
+* `references.bib`, `references.json`, `references.yaml`, `references.bibtex`
+
+To customize the global bibliography file names, set the `PANDOC_GLOBAL_BIB_FILES` setting in your Pelican settings file:
```
I would prefer that this variable be named `PANDOC_GLOBAL_BIBLIOGRAPHIES`

nandac commented on README.md:
````diff
@@ -181,9 +181,31 @@ Without these settings, citations will not be processed by the plugin.
+To customize the global bibliography file names, set the `PANDOC_GLOBAL_BIB_FILES` setting in your Pelican settings file:
+
+```python
+PANDOC_GLOBAL_BIB_FILES = ["my_bib", "global_refs", "shared_bibliography"]
````
Could you please format this like this so that it follows the standard used in the rest of the README:
```
PANDOC_GLOBAL_BIBLIOGRAPHIES = [
    "my_bib",
    "global_refs",
    "shared_bibliography"
]
```

nandac commented on pelican/plugins/pandoc_reader/pandoc_reader.py:
```diff
@@ -390,17 +390,24 @@ def _check_if_toc(arguments):
                 bib_name = ".".join([filename, extension])
                 if bib_name in files:
                     bib_files.append(os.path.join(root, bib_name))
+                for global_name in global_bib_names:
```
I am confused with this code. Could you tell me what the path to the default bibliographies is?

A centralized path for all default bibliographies would make more sense to me.

nandac commented on pelican/plugins/pandoc_reader/test/markdown/_bibliography.bib:
```diff
@@ -0,0 +1,16 @@
+author = {Charlie Wood},
+date = {2019-07-11},
+urldate = {2020-11-12}
+} 
```
Please add the new line.

nandac commented on pelican/plugins/pandoc_reader/test/markdown/bibliography.bib:
```diff
@@ -0,0 +1,17 @@
+author = {Rafael Alves Batista and Joel Primack},
+date = {circa 2017},
+urldate = {2020-11-12}
+} 
```
Please add the new line.

nandac commented on pelican/plugins/pandoc_reader/test/markdown/content_with_both_bibs.bib:
```diff
@@ -0,0 +1,7 @@
+author = {Andrew Zimmerman Jones},
+date = {2019-03-02},
+urldate = {2020-11-12}
+} 
```
Please add the new line.

nandac commented on pelican/plugins/pandoc_reader/test/markdown/my_bib.bib:
```diff
@@ -0,0 +1,16 @@
+author = {Davide Castelvecchi},
+date = {2016-01-05},
+urldate = {2020-11-12}
+} 
```
Please add the new line.

nandac commented:
@filipaldi, I have added some preliminary review comments. Kindly also fix the linting issues when you get a chance.

I like the addition of the `pelicanconf_example.py` file, but as it does not apply directly to the plugin, I am not sure if this is exactly the right place for it.
